packages/SwingSet/src/vats/vattp/vat-vattp.js

```diff
@@ -16,181 +23,255 @@ import { Far } from '@endo/marshal';
-      assert(!remotes.has(name), X`already have remote ${name}`);
-      const r = getRemote(name);
-      const transmitter = Far('transmitter', {
-        transmit(msg) {
-          const o = r.outbound;
-          const num = o.highestAdded + 1;
-          // console.debug(`transmit to ${name}[${num}]: ${msg}`);
-          D(mailbox).add(name, num, msg);
-          o.highestAdded = num;
-        },
-      });
-      const setReceiver = Far('receiver setter', {
-        setReceiver(newReceiver) {
-          assert(!r.inbound.receiver, X`setReceiver is call-once`);
-          r.inbound.receiver = newReceiver;
-        },
-      });
+      assert(!mailboxes.has(name), X`already have remote ${name}`);
+      const { transmitter, setReceiver } = provideMailbox(name);
       return harden({ transmitter, setReceiver });
     },
 
     deliverInboundMessages(name, newMessages) {
-      const i = getRemote(name).inbound;
-      newMessages.forEach(m => {
-        const [num, body] = m;
-        if (num > i.highestDelivered) {
-          // TODO: SO() / sendOnly()
-          // console.debug(`receive from ${name}[${num}]: ${body}`);
-          E(i.receiver).receive(body);
-          i.highestDelivered = num;
-          D(mailbox).ackInbound(name, num);
-        }
-      });
+      // TODO: Stop silently creating mailboxes.
+      // https://github.com/Agoric/agoric-sdk/issues/5824
+      const { inbound } = provideMailbox(name);
+      inbound.deliverMessages(newMessages);
     },
 
     deliverInboundAck(name, ack) {
-      const o = getRemote(name).outbound;
-      let num = o.highestRemoved + 1;
-      while (num <= o.highestAdded && num <= ack) {
-        D(mailbox).remove(name, num);
-        o.highestRemoved = num;
-        num += 1;
-      }
+      // TODO: Stop silently creating mailboxes.
+      // https://github.com/Agoric/agoric-sdk/issues/5824
+      const { inbound } = provideMailbox(name);
+      inbound.deliverAck(ack);
     },
+  };
 
-    makeNetworkHost,
-  });
+  // Network comms pattern:
+  //
+  // A:
+  //
+  //   E(agoric.ibcport[0]).addListener({
+  //     onAccept: () => {
+  //       const { host, handler } = await E(agoric.vattp).makeNetworkHost('ag-chain-B', comms);
+  //       const helloAddress = await E(host).publish(hello);
+  //       // helloAddress = '/alleged-chain/${chainID}/egress/${clistIndex}'
+  //       return handler;
+  //     },
+  //   });
+  //
+  // B:
+  //
+  //   const { host, handler } = await E(agoric.vattp).makeNetworkHost('ag-chain-A', comms);
+  //   E(agoric.ibcport[1]).connect('/ibc-port/portADDR/ordered/vattp-1', handler);
+  //   E(E(host).lookup(helloAddress)).hello();
+
+  const nextConnectionNumberBaggageKey = 'networkHostNextConnectionNumber';
+  let nextConnectionNumber = provide(
+    baggage,
+    nextConnectionNumberBaggageKey,
+    () => 7n,
+  );
+  // TODO: Should this collection be Weak (non-iterable)?
+  const networkHostNames = provideDurableSetStore(baggage, 'networkHostNames');
```

Comment:
It's indexed by string, so I think it can't be a WeakStore (my reading of the Store docs say that only Remotables can be keys, which aligns with a normal `new WeakMap()` only accepting `Object`s as keys).

Reply:
According to [A Taxonomy of Stores](https://github.com/Agoric/agoric-sdk/blob/0e8644a34f996242935d4a525c631ae32e1da462/packages/store/docs/store-taxonomy.md#user-content-weak-vs-strong), the relevant characteristic of weak collections is that they are not iterable—but they "_may_" allow forgeable keys (with the understanding that such entries can never be GC'd). So it seems like a WeakSetStore of strings is possible here, but it wouldn't support any automatic reduction of stored data.

Reply:
yeah, I think our current implementation doesn't:

```js
import { makeScalarWeakMapStore } from '@agoric/store';
test('foo', async t => {
  const s = makeScalarWeakMapStore();
  s.init('s', 'str');
  t.pass();
});
```

gets me:
```
REJECTED from ava test: (Error#1)
Error#1: Only remotables can be keys of scalar WeakMapStores: s
  at assertKVOkToAdd (packages/store/src/stores/scalarWeakMapStore.js:117:5)
  at Alleged: scalar WeakMapStore of "key".init (packages/store/src/stores/scalarWeakMapStore.js:45:7)
  at packages/SwingSet/test/test-vat-timer.js:763:5
  at async Promise.all (index 1)
```